### PR TITLE
Feat/synonyms

### PR DESCRIPTION
**Pull request by ShahAnuj2610**
add synonyms category and plan validations

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -187,6 +189,9 @@ func main() {
 	if FeatureSearchSettings != "" && FeatureSearchSettings == "true" {
 		util.SetFeatureSearchSettings(true)
 	}
+	if FeatureSynonyms == "true" {
```
Comment (lakhansamani) [inline main.go:192]: Also, add a check for an empty string.

**Review by lakhansamani — CHANGES_REQUESTED** [review]

**Comment by lakhansamani** [comment]
Please resolve conflicts here.

File: main.go (hunk 1 above)
Comment (ShahAnuj2610) [reply main.go:192]: this is not needed. We're directly comparing a string. So 
`"" == "true"` would result to `false`, which is what we wanted

File: util/feature.go [hunk 2]
```diff
@@ -77,3 +77,16 @@ func GetFeatureSearchRelevancy() bool {
 func SetFeatureSearchRelevancy(val bool) {
 	featureSearchRelevancy = val
 }
+
+// Feature synonyms
+var featureSynonyms bool
+
+// GetFeatureSynonyms returns the featureSearchSettings
```
Comment (bietkul) [inline util/feature.go]: @ShahAnuj2610 Fix description

File: util/feature.go [hunk 3]
```diff
@@ -77,3 +77,16 @@ func GetFeatureSearchRelevancy() bool {
...
+func GetFeatureSynonyms() bool {
+	return featureSynonyms
+}
+
+// SetFeatureSynonyms sets the featureSearchSettings
```
Comment (bietkul) [inline util/feature.go]: @ShahAnuj2610 Fix description

**Review by bietkul — CHANGES_REQUESTED** [review]

**Review by bietkul — APPROVED** [review]

**Pull request merged**